### ISSUE TITLE
[Go] shouldn't use XDG_CACHE_HOME for reading kubeconfigs

### ISSUE THREAD

**Issue by ahmetb** [ahmetb/kubectx#254]
bash based implementations used something like `KUBECTX="${XDG_CACHE_HOME:-$HOME/.kube}/kubectx"` to determine where the "state files" are stored.

XDG_CACHE_HOME is good for that kinda stuff.

However with the Go implementations, that path is also used for determining where the $KUBECONFIG file is at by default –which is not correct. client-go default config loading rules don't read XDG_ env vars.

**Comment by josegonzalez** [comment]
I think this can be closed.

**Comment by aslafy-z** [comment]
I don't think this issue should be closed: The Go implementation stills push cache files to the `$HOME/.kube` directory: https://github.com/ahmetb/kubectx/blob/master/cmd/kubectx/state.go#L27-L33.
